Bump version workflow failing

### Issue description
Bump version workflow in master is failing due to issue with the step where `bump2version` is installed on the runner.
This issue is seen in Ubuntu24.04.  


### Error log

See failed workflow run in https://github.com/tudat-team/tudatpy/actions/runs/11339725923/job/31534995872 

```
Run pip install bump2version
error: externally-managed-environment

× This environment is externally managed
╰─> To install Python packages system-wide, try apt install
    python3-xyz, where xyz is the package you are trying to
    install.
    
    If you wish to install a non-Debian-packaged Python package,
    create a virtual environment using python3 -m venv path/to/venv.
    Then use path/to/venv/bin/python and path/to/venv/bin/pip. Make
    sure you have python3-full installed.
    
    If you wish to install a non-Debian packaged Python application,
    it may be easiest to use pipx install xyz, which will manage a
    virtual environment for you. Make sure you have pipx installed.
    
    See /usr/share/doc/python3.12/README.venv for more information.

note: If you believe this is a mistake, please contact your Python installation or OS distribution provider. You can override this, at the risk of breaking your Python installation or OS, by passing --break-system-packages.
hint: See PEP 668 for the detailed specification.
Error: Process completed with exit code 1.
```


### Fix

Install python packages in a python virtual environment instead of system wide installation.

--- a/.github/workflows/bump_dev_version_nightly.yml
+++ b/.github/workflows/bump_dev_version_nightly.yml
@@ -18,7 +18,7 @@ name: Bump dev version in tudatpy and tudatpy-feedstock if new commit in last 24
 
 on:
     schedule:
-        - cron: '0 5 * * *'  # This triggers the workflow daily at 5:00 AM UTC. 
+        - cron: '0 5 * * *'  # This triggers the workflow daily at 5:00 AM UTC.
     workflow_dispatch:       # This facilitates manual triggering of the workflow from the Actions tab of the repository in case the scheduled run fails.
 
 
@@ -63,17 +63,16 @@ jobs:
         with:
           ref: develop
 
-      - name: Install bump2version
-        run: pip install bump2version
-
-      - name: Bump version
-        id: bump_version
+      - name: Install bump2version and run bumpversion
         run: |
-          git config --global user.email "actions@github.com"
-          git config --global user.name "GitHub Actions"
-          bump2version dev --config-file .bumpversion.cfg --verbose
+            python3 -m venv venv
+            source venv/bin/activate
+            pip install bump2version
+            git config --global user.email "actions@github.com"
+            git config --global user.name "GitHub Actions"
+            bump2version dev --config-file .bumpversion.cfg --verbose
         env:
-          GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
+            GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
 
       - name: Push tag to tudatpy repository
         run: git push origin --follow-tags
@@ -94,14 +93,16 @@ jobs:
             token: ${{ secrets.BUMP_VERSION_NIGHTLY }}
             ref: develop
 
-        - name: Install bump2version
-          run: pip install bump2version
-
-        - name: Bump version on feedstock
+        - name: Install bump2version and run bumpversion
           run: |
-              git config --global user.email "actions@github.com"
-              git config --global user.name "GitHub Actions"
-              bump2version dev --config-file .bumpversion.cfg --verbose
+               python3 -m venv venv
+               source venv/bin/activate
+               pip install bump2version
+               git config --global user.email "actions@github.com"
+               git config --global user.name "GitHub Actions"
+               bump2version dev --config-file .bumpversion.cfg --verbose
+          env:
+              GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
 
         - name: Reset build number in meta.yml to 0
           shell: bash

--- a/.github/workflows/bump_dev_version_nightly.yml
+++ b/.github/workflows/bump_dev_version_nightly.yml
@@ -71,8 +71,6 @@ jobs:
             git config --global user.email "actions@github.com"
             git config --global user.name "GitHub Actions"
             bump2version dev --config-file .bumpversion.cfg --verbose
-        env:
-            GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
 
       - name: Push tag to tudatpy repository
         run: git push origin --follow-tags
@@ -101,8 +99,6 @@ jobs:
                git config --global user.email "actions@github.com"
                git config --global user.name "GitHub Actions"
                bump2version dev --config-file .bumpversion.cfg --verbose
-          env:
-              GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
 
         - name: Reset build number in meta.yml to 0
           shell: bash

